src/it/java/com/google/checkstyle/test/chapter4formatting/rule412nonemptyblocks/NonemptyBlocksKRStyleTest.java

```diff
@@ -0,0 +1,94 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code and other text files for adherence to a set of rules.
+// Copyright (C) 2001-2024 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.google.checkstyle.test.chapter4formatting.rule412nonemptyblocks;
+
+import org.junit.jupiter.api.Test;
+
+import com.google.checkstyle.test.base.AbstractGoogleModuleTestSupport;
+
+// -@cs[AbbreviationAsWordInName] Test should be named as its chapter name.
+public class NonemptyBlocksKRStyleTest extends AbstractGoogleModuleTestSupport {
```

Comment:
`NonemptyBlocksKrStyleTest`
Please respect, Abbreviation rules of GOOGLE style.

Reply:
The section name is: `Nonempty blocks: K & R Style`

Here `K` and `R` both different words, that's why they're in capital.


Based on the implementation of XdocPagesTest validation, we need to use the `NonemptyBlocksKRStyleTest`, if we change the name implementation it'd affect other names too.

Reply:
But we violate Google style on Abbreviations.
Let's update XdocPagesTest validation to change remove ` & ` , to let it become KR as abbreviation, and later on end up as 'Kr'

Reply:
https://checkstyle.sourceforge.io/styleguides/google-java-style-20180523/javaguide.html#s5.3-camel-case

Reply:
done

https://github.com/checkstyle/checkstyle/blob/9c780b1ce791087f3a9963561f315e3621aa93e8/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java#L2010-L2016

Reply:
removed the suppression